File: leetcode_problems/median_of_Two_Sorted_Arrays.c
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

double findMedianSortedArrays(int *nums1, int nums1Size, int *nums2, int nums2Size);
void printArray(int *nums, int numsSize);

double findMedianSortedArrays(int *nums1, int nums1Size, int *nums2, int nums2Size)
{
    const int result_len = nums1Size + nums2Size;
    int result[result_len];
    int p1 = 0;
    int p2 = 0;
    int i = 0;
    while (true && i < 5)
    {
        if (p1 == nums1Size)
        {
            // printf("Break 1\n");
            const ssize_t size = sizeof(int) * (nums2Size - p2);
            memcpy(&result[i], &nums2[p2], size);
            break;
        }
        else if (p2 == nums2Size)
        {
            // printf("Break 2\n");
            const ssize_t size = sizeof(int) * (nums1Size - p2);
            memcpy(&result[i], &nums1[p1], size);
            break;
        }
        if (nums1[p1] <= nums2[p2])
        {
            result[i] = nums1[p1];
            ++p1;
        }
        else
        {
            result[i] = nums2[p2];
            ++p2;
        }
        ++i;
    }
    // printArray(result, result_len);
    if (result_len % 2 == 0)
    {
        double val1 = result[(int)(result_len / 2) - 1];
        double val2 = result[(int)(result_len / 2)];
        double ret_val = (val1 + val2)/2;
        // printf("1. Ret vals: %f %f \t Median %f\n", val1, val2, ret_val);
        return ret_val;
    }
    else
    {
        double ret_val = result[(int)(result_len / 2)];
        // printf("2. Ret vals: %f\n", ret_val);
        return ret_val;
    }
    return EXIT_FAILURE;
}

int main(void)
{
    // int nums1[] = {};
    // int nums2[] = {2,3,4,5,6};
    int nums1[] = {0,0,0,0,0};
    int nums2[] = {-1,0,0,0,0,0,1};
    double a = findMedianSortedArrays(nums1, (int)sizeof(nums1) / sizeof(int), nums2, (int)sizeof(nums2) / sizeof(int));
    printf("Median: %f\nLen Array: %d\n", a, (int)((sizeof(nums1) + sizeof(nums2)) / sizeof(int)));
    return 0;
}

void printArray(int *nums, int numsSize)
{
    printf("Array: ");
    for (int i = 0; i < numsSize; i++)
    {
        printf("%d ", nums[i]);
    }
    printf("\n");
}

File: leetcode_problems/median_of_Two_Sorted_Arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1, nums2) -> float:
        result = nums1 + nums2
        result = sorted(result)
        result_len = len(result)
        if (result_len % 2):
            return result[int(result_len / 2)]
        else:
            result = result[int(result_len / 2)-1:int(result_len / 2)+1]
            return(sum(result)/2)
    
    def merge(self, nums1, nums2):
        p1 = 0
        p2 = 0
        n = max(len(nums1), len(nums2))
        result = []
        if (not len(nums1)):
            return nums2
        if (not len(nums2)):
            return nums1
        return result

def main():
    sol=Solution()
    nums1=[1, 3, 4, 6, 8, 9]
    nums2=[2, 7]
    a = sol.findMedianSortedArrays(nums1, nums2)
    print(a)

if __name__ == "__main__":
    main()
